academy/0.0-B9lab-Blockchains/1_blockchain.md

```diff
@@ -0,0 +1,217 @@
+
+Consequently, it is generally not possible for two parties to exchange value online without involving a trusted third party to handle the settlement process and update their ledgers, as well as the account balances.
+
+At a high level, Bitcoin solves the double-spending problem by replacing the trusted, central ledger-keepers with a decentralized and distributed ledger that is maintained by a large network of ledger-keepers. Each member of the network has an exact replica of the ledger, and no one can update the ledger without establishing consensus with the other ledger-keepers.
+
+![Double-spending problem due to insufficient funds](images/00_04_double_spending-01.png)
+
+It is as though each transaction is observed by a *large crowd of witnesses* who reach consensus about proposed changes. The crowd prohibits events that should not occur, such as spending the same funds twice (i.e. double-spending).
+
+Bitcoin and its underlying technology convincingly demonstrate that a network of participants that do not necessarily trust each other can achieve consensus about the validity of a transaction, its history, and the resulting state of the ledger. This is interesting, because simple account balance ledgers and protocols to move funds are far from the only use cases for distributed consensus.
+
+### How does blockchain work?
+
+Imagine you want to **retain and monitor changes** to a file, for example a log file. Now, imagine you also want to verify an unbroken history of all changes ever made to the file. How can you proceed?
+
+A well-understood solution uses **cryptographic [hash functions](https://en.wikipedia.org/wiki/Cryptographic_hash_function)**. Let us briefly introduce this concept in case you are unfamiliar with them.
+
+The ideal cryptographic hash function has **five main properties**:
+
+* **Deterministic:** the same message always results in the same hash.
+* **Fast:** the hash value for any given message is computed quickly.
+* **Resistant:** it is not feasible to generate a message from its hash value except by trying all possible messages.
+* **Uncorrelated:** a small change to a message alters the hash value so extensively that the new value shows no relation to the old.
+* **Collision-resistant:** it is infeasible to find two different messages with the same hash value.
+
+<HighlightBox type="tip">
+
+You can see hashing in action to get the feel for it here: [http://onlinemd5.com/"](http://onlinemd5.com/). As you type into the text box, the hash updates automatically. Even a minuscule change to the input creates a completely different hash. Try it out!
```

Comment:
Don't we mention elsewhere in the content that MD5 has been broken? It should be close by if missing. I think there is value in telling that a hash can be broken. Not sure it would be better told if we started with one that is not broken.

Reply:
I've change the example to https://www.browserling.com/tools/all-hashes